This patch is based on the commit described below, taken from the upstream
time64 pull request and modified for use in the Debian package by Peter
Michael Green.

commit eebc18690dc2a2173737251f09c874ef39424395
Author: Ola x Nilsson <olani@axis.com>
Date:   Mon Mar 27 14:15:18 2023 +0200

    gnu: Update struct msqid_ds for 64-bit time

Index: libc/src/unix/linux_like/linux/gnu/b32/arm/mod.rs
===================================================================
--- libc.orig/src/unix/linux_like/linux/gnu/b32/arm/mod.rs
+++ libc/src/unix/linux_like/linux/gnu/b32/arm/mod.rs
@@ -84,10 +84,13 @@ s! {
     pub struct msqid_ds {
         pub msg_perm: crate::ipc_perm,
         pub msg_stime: crate::time_t,
+        #[cfg(not(gnu_time64_abi))]
         __glibc_reserved1: c_ulong,
         pub msg_rtime: crate::time_t,
+        #[cfg(not(gnu_time64_abi))]
         __glibc_reserved2: c_ulong,
         pub msg_ctime: crate::time_t,
+        #[cfg(not(gnu_time64_abi))]
         __glibc_reserved3: c_ulong,
         __msg_cbytes: c_ulong,
         pub msg_qnum: crate::msgqnum_t,
Index: libc/src/unix/linux_like/linux/gnu/b32/mips/mod.rs
===================================================================
--- libc.orig/src/unix/linux_like/linux/gnu/b32/mips/mod.rs
+++ libc/src/unix/linux_like/linux/gnu/b32/mips/mod.rs
@@ -169,21 +169,9 @@ s! {
 
     pub struct msqid_ds {
         pub msg_perm: crate::ipc_perm,
-        #[cfg(target_endian = "big")]
-        __glibc_reserved1: c_ulong,
         pub msg_stime: crate::time_t,
-        #[cfg(target_endian = "little")]
-        __glibc_reserved1: c_ulong,
-        #[cfg(target_endian = "big")]
-        __glibc_reserved2: c_ulong,
         pub msg_rtime: crate::time_t,
-        #[cfg(target_endian = "little")]
-        __glibc_reserved2: c_ulong,
-        #[cfg(target_endian = "big")]
-        __glibc_reserved3: c_ulong,
         pub msg_ctime: crate::time_t,
-        #[cfg(target_endian = "little")]
-        __glibc_reserved3: c_ulong,
         __msg_cbytes: c_ulong,
         pub msg_qnum: crate::msgqnum_t,
         pub msg_qbytes: crate::msglen_t,
Index: libc/src/unix/linux_like/linux/gnu/b32/powerpc.rs
===================================================================
--- libc.orig/src/unix/linux_like/linux/gnu/b32/powerpc.rs
+++ libc/src/unix/linux_like/linux/gnu/b32/powerpc.rs
@@ -145,11 +145,8 @@ s! {
 
     pub struct msqid_ds {
         pub msg_perm: crate::ipc_perm,
-        __glibc_reserved1: c_uint,
         pub msg_stime: crate::time_t,
-        __glibc_reserved2: c_uint,
         pub msg_rtime: crate::time_t,
-        __glibc_reserved3: c_uint,
         pub msg_ctime: crate::time_t,
         __msg_cbytes: c_ulong,
         pub msg_qnum: crate::msgqnum_t,
Index: libc/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
===================================================================
--- libc.orig/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
+++ libc/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
@@ -189,11 +189,8 @@ s! {
     pub struct msqid_ds {
         pub msg_perm: crate::ipc_perm,
         pub msg_stime: crate::time_t,
-        __glibc_reserved1: c_ulong,
         pub msg_rtime: crate::time_t,
-        __glibc_reserved2: c_ulong,
         pub msg_ctime: crate::time_t,
-        __glibc_reserved3: c_ulong,
         __msg_cbytes: c_ulong,
         pub msg_qnum: crate::msgqnum_t,
         pub msg_qbytes: crate::msglen_t,
